doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,183 @@
+connections to be received.  Ideally, most scripts spend most of their time
+here.
+
+## `setImmediate()`:
+
+`setImmediate()` is actually a special timer that runs in a separate phase of
+the event loop.  It uses a libuv API that schedules callbacks to execute after
+the poll phase has completed.
+
+Generally, as the code is executed, the event loop will eventually hit the
+`poll` phase where it will wait for an incoming connection, request, etc.
+However, after a callback has been scheduled with `setImmediate()`, at the
+start of the poll phase, a check will be run to see if there are any callbacks
+waiting.  If there are none waiting, the poll phase will end and continue to
+the `setImmediate` callback phase.
+
+### `setImmediate()` vs `setTimeout()`
+
+How quickly a `setImmediate()` callback is executed is only limited by how
+quickly the event loop can be processed whereas a timer won't fire until the
+number of milliseconds passed have elapsed.
+
+The advantage to using `setImmediate()` over `setTimeout()` is that the lowest
+value you may set a timer's delay to is 1 ms (0 is coerced to 1) which doesn't
+seem like much time to us humans, but it's actually pretty slow compared to
+how quickly `setImmediate()` can execute — the event loop operates on the
+microsecond scale (1 ms = 1000 µs.)
+
+## `process.nextTick()`:
+
+### Understanding `process.nextTick()`
+
+You may have noticed that `process.nextTick()` was not displayed in the
+diagram, even though its a part of the asynchronous API.  This is because
+`process.nextTick()` is not technically part of the event loop.  Instead, it
+is executed at the end of each phase of the event loop.
+
+Looking back at our diagram, any time you call `process.nextTick()` in a given
+phase, all callbacks passed to `process.nextTick()` will be resolved before
+the event loop continues. This can create some bad situations because **it
+allows you to "starve" your I/O by making recursive `process.nextTick()`
+calls.**  which prevents the event loop from reaching the poll phase.
+
+### Why would that be allowed?
+
+Why would something like this be included in Node.js?  Part of it is a design
+philosophy where an API should always be asynchronous even where it doesn't
+have to be.  Take this code snippet for example:
+
+```js
+function apiCall (arg, callback) {
+  if (typeof arg !== 'string')
+    return process.nextTick(callback,
+      new TypeError('argument should be string'));
+}
+```
+
+The snippet does an argument check and if it's not correct, it will pass the
+error to the callback.  The API updated fairly recently to allow passing
+arguments to `process.nextTick()` allowing it to take any arguments passed
+after the callback to be propagated as the arguments to the callback so you
+don't have to nest functions.
+
+What we're doing is passing an error back to the user.  As far as the _event
+loop_ is concerned, this happens **synchronously**. However, as far as the
```

Comment:
I think he means that queuing callbacks with `nextTick` will not allow the event loop to proceed from say `uv__io_poll()` to `uv__run_check()`. So from the `uv_run()` perspective it would be synchronous. Have a better way of describing that?